nan_json.h

```diff
@@ -0,0 +1,160 @@
+/*********************************************************************
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2017 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#ifndef NAN_JSON_H_
+#define NAN_JSON_H_
+
+#if NODE_MODULE_VERSION < NODE_0_12_MODULE_VERSION
+#define NAN_JSON_H_NEED_PARSE 1
+#else
+#define NAN_JSON_H_NEED_PARSE 0
+#endif  // NODE_MODULE_VERSION < NODE_0_12_MODULE_VERSION
+
+#if NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+#define NAN_JSON_H_NEED_STRINGIFY 0
+#else
+#define NAN_JSON_H_NEED_STRINGIFY 1
+#endif  // NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+
+class JSON {
+ public:
+  JSON() {
+#if NAN_JSON_H_NEED_PARSE + NAN_JSON_H_NEED_STRINGIFY
+    Nan::MaybeLocal<v8::Value> maybe_global_json = Nan::Get(
+      Nan::GetCurrentContext()->Global(),
+      Nan::New("JSON").ToLocalChecked()
+    );
+
+    if (!maybe_global_json.IsEmpty()) {
```

Comment:
So now, if the user deliberately screws with the global `JSON` object *and* compiles with `NDEBUG` defined, bad things will happen.    It makes me wonder if I should have just ADDED the `assert`s rather than replacing the checks with `assert`s, but that would be redundant.

What would you like to see done, here?  Should I leave these `assert`s as they are?  Should I bring back the checks that I previously had, but after the `assert`s ?   I don't think we should throw any exceptions, but maybe that is a better path?